### ISSUE TITLE
Compilation errors with Hydra on master

### ISSUE THREAD

**Issue by iravid** [zio/zio#2664]
Getting these errors:
```
[error] /Users/iravid/Development/personal/scalaz-zio/test/shared/src/main/scala/zio/test/package.scala:24:8: object TestEnvironment is not a member of package zio.test.environment
[error] import zio.test.environment.{ testEnvironmentManaged, TestClock, TestConsole, TestEnvironment, TestRandom, TestSystem }
[error] ^
[error] /Users/iravid/Development/personal/scalaz-zio/test/shared/src/main/scala/zio/test/package.scala:325:37: not found: type TestEnvironment
[error] val defaultTestRunner: TestRunner[TestEnvironment, Any, String, Any, Any] =
[error] ^
[warn] /Users/iravid/Development/personal/scalaz-zio/test/shared/src/main/scala/zio/test/package.scala:24:79: Unused import
[warn] import zio.test.environment.{ testEnvironmentManaged, TestClock, TestConsole, TestEnvironment, TestRandom, TestSystem }
[warn] ^
[error] /Users/iravid/Development/personal/scalaz-zio/test/shared/src/main/scala/zio/test/package.scala:325:26: erroneous or inaccessible type
[error] val defaultTestRunner: TestRunner[TestEnvironment, Any, String, Any, Any] =
[error] ^
[warn] sbt-api: Unhandled type class scala.reflect.internal.Types$ErrorType$ : <error>
```

From a chat with @dragos:
> 
> Hi Itamar,
> 
> Thanks for reporting it! I can reproduce the issue, and unfortunately it's a very tricky corner case. This is something that can fail with the regular Scala compiler as well in incremental builds, but it's less likely than with Hydra. Here's the gist:
> 
> There's a package object zio.test.environment and this one extends zio.test.environment.PlatformSpecific. Notice that this is a cycle: in order to define what's inside package environment, the compiler needs to resolve and extends PlatformSpecific... which is itself defined inside package environment! This shouldn't work, and it's through an accident and a lucky ordering in which files and directories are opened that it does, sometimes. There's a discussion at scala-dev#441 and a PR to remove package objects with inheritance here.
> 
> The workaround is simple: move PlatformSpecific one level up, inside package zio.test. I tried it out and it works. Since this will likely go away or be deprecated in 2.14 (and not supported in Dotty), I think it's not worth a lot of effort in trying to find a fix (which will likely cover only some of all real world cases. To give an idea, already spent a lot of effort trying to make this work in as many cases as possible.I'm pretty sure that even the current codebase  would compile fine if they were all part of the same source tree, instead of being split in `src/shared` and `src/jvm` and `src/js`.
> 
> Please let me know if the workaround is acceptable (and works for you as well).

I'll take care of this if no-one gets to it before me.

**Comment by vasilmkd** [comment]
I think this is also the main hindrance when trying to import the zio codebase into IntelliJ. We might need to fix this in other packages as well, i.e. `streams`.

**Comment by adamgfraser** [comment]
Looks like I caused this so I will take care of it now.

**Comment by iravid** [comment]
@adamgfraser #2667 did not fix the issue, so reopening this.

**Comment by adamgfraser** [comment]
Okay. At this point there is no difference between the implementation in ZIO Test and the ones in Core that I can tell. Possibly there is some implementation detail of class loading? We should probably change them all to remove the circularity to avoid this. Will work on later today.

**Comment by iravid** [comment]
Is it possible to move `PlatformSpecific` up to `zio.test` like Iulian suggested?

**Comment by adamgfraser** [comment]
I think so.

**Comment by iravid** [comment]
Ok. Let me know if I can help.

**Comment by iravid** [comment]
Fixed in #2673.
